crates/compiler/mono/src/ir.rs

```diff
@@ -1875,6 +1849,12 @@ pub enum Expr<'a> {
         union_layout: UnionLayout<'a>,
         index: u64,
     },
+    UnionFieldPtrAtIndex {
+        structure: Symbol,
+        tag_id: TagIdIntType,
+        union_layout: UnionLayout<'a>,
+        index: u64,
+    },
```

Comment:
Maybe a doc comment that this is only used for TRMC, and must always point into the field that is a recursion pointer of the union?

Reply:
the second bit does not technically need to be true. it is true in practice, right now, but in the future we might find different uses for this expr.